ee/clickhouse/queries/session_recordings/clickhouse_session_recording.py

```diff
@@ -26,7 +26,7 @@ def _query_recording_snapshots(self) -> List[SessionRecordingEvent]:
                 window_id=window_id,
                 distinct_id=distinct_id,
                 timestamp=timestamp,
-                snapshot_data=json.loads(snapshot_data),
+                snapshot_data=try_read_from_object_storage(session_id, snapshot_data),
```

Comment:
Might be good to dive into this (maybe separately...)

naively we send the data across the network to "a service" and then read it across the network from the service

Assuming they have similar latency there's no impact.

----

Less naively they're not the same...

Existing we read a set of events from ClickHouse. One network connection carries all the data 

vs 

we read a set of events from ClickHouse, one network connection reads everything but snapshot data may not be on each event, and for each event that doesn't have snapshot data we read that from object_storage

More network connections but (roughly) the same amount of data

-----

Which is why it's timed (although not working :)) above. My guess is that S3 is fast enough and minio doesn't add enough latency that it's a good enough step forward and lets us reduce storage in ClickHouse...

----

Or am I either wrong or misunderstanding you :)

Reply:
I had something else in mind - I originally pitched to @rcmarron that we should _compress_ the whole session recording after the fact and upload it to s3 rather than do this per individual events (of which there can be thousands). This will yield a nice compressed payload user can download at once from the backend, speeding up everything.

However this was not the route you took and at this point this feedback detracts rather than adds to this review. Resolving.